bin/lib/utils/utils.js

```diff
@@ -153,16 +157,43 @@ function processEnv(name, config={}){
 	}
 
 	if (config.hasOwnProperty('validateInteger')) {
-		value = parseInt(value);
-		if (! config.validateInteger(value)) {
-			throw new Error(config.errorMsg);
+		const configErrMsg = `The configuration for env param ${name} should be Boolean true (to ensure conversion to Integer) or a function defintion (to ensure conversion to Integer, and then validate the integer value)`;
+		// accept 'true' but not 'false', or a fn definition
+		const configType = typeof config.validateInteger;
+		value = parseInt(value); // always convert to integer
+
+		if (configType === "boolean"){ // accept 'true' but not 'false'
+			if (config.validateInteger === false) {
+				throw new Error(configErrMsg);
+			}
+		} else if (configType === 'function') {
+			if(! config.validateInteger(value)) {
+					throw new Error(config.errorMsg);
+			}
+		}	else {
+			throw new Error(configErrMsg);
 		}
 	} else if(config.hasOwnProperty('validateJson')){
+		const configErrMsg = `The configuration for env param ${name} should be Boolean true (to ensure parsing as JSON) or a function defintion (to ensure parsing JSON, and then validate the parsed value)`;
+		// accept 'true' but not 'false', or a fn definition
+		const configType = typeof config.validateJson;
 		try {
 			value = JSON.parse(value);
 		} catch(e) {
 			throw new Error(config.errorMsg + ': error during parsing JSON: ' + e.message);
 		}
+
```

Comment:
I feel this whole if/else statement could be tidied eg. first line `if (configType === 'boolean' && !config.validateJson)`

Reply:
it is logically correct. Not sure that compacting it would still cover all the edge cases.